crates/services/txpool_v2/src/spent_inputs.rs

```diff
@@ -0,0 +1,283 @@
+use fuel_core_types::{
+    fuel_tx::{
+        Input,
+        TxId,
+        UtxoId,
+    },
+    fuel_types::Nonce,
+};
+use lru::LruCache;
+use std::{
+    collections::HashMap,
+    num::NonZeroUsize,
+};
+
+#[derive(Clone, Copy, Hash, PartialEq, Eq, Debug)]
+enum InputKey {
+    Utxo(UtxoId),
+    Message(Nonce),
+}
```

Comment:
possible to store references here instead?
```suggestion
enum InputKey<'a> {
    Utxo(&'a UtxoId),
    Message(&'a Nonce),
}
```

Reply:
It is not possible, because transactions live shorter than this cache=)